packages/core/server/index.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable global-require, no-console, import/order */
+const {
+  API_ROOT_URL,
+  API_ORIGIN,
+} = process.env;
+
+// Start monitor service as early as possible.
+const getService = require('../services/monitorService');
+getService().start();
+
+// Shim some browser-only global variables.
+require('../utils/shimWindow');
+
+const express = require('express');
+const { createProxyMiddleware } = require('http-proxy-middleware');
+const cookiesMiddleware = require('universal-cookie-express');
+const getConfigField = require('../utils/getConfigField');
+const { getConfigArray } = require('../utils/getConfigValue');
+
+const getLogService = require('../services/logService');
+const customizeRedirect = require('./customizeRedirect');
+const log = getLogService('irving:server');
+const app = express();
+
+// Cache-related endpoints.
+require('./cache')(app);
+
+// Set view engine.
+app.set('view engine', 'ejs');
+
+// Run all customize server functions.
+const irvingServerMiddleware = getConfigField('customizeServer');
+irvingServerMiddleware.forEach((middleware) => middleware(app));
+
+// Set up a reusable proxy for responses that should be served directly.
+const proxyPassthrough = getConfigArray('proxyPassthrough');
+const passthrough = createProxyMiddleware({
+  changeOrigin: true,
+  followRedirects: true,
+  secure: 'development' !== process.env.NODE_ENV,
+  // @todo make this not specific to WP eventually.
```

Comment:
Doesn't the use of `API_ORIGIN` takes care of the WP specificity?

Reply:
I guess so yeah, so maybe I can remove this todo.